### ISSUE TITLE
"no-misspelled-properties" rule treats certain nested properties preceded by control directives as spelling errors

### ISSUE THREAD

**Issue by EricDunsworth** [sasstools/sass-lint#1158]
This is a simplified description of an issue I ran across in wet-boew/theme-gcwu-fegc#251.

**What version of Sass Lint are you using?**
1.12.1 (via grunt-sass-lint 0.2.4)

**Please include any relevant parts of your configuration**
```
rules:
  indentation:
    - 1
    - size: tab
  no-misspelled-properties: 1
```

**What did you do? Please include the actual source code causing the issue.**
```
.test {
	background: {
		@if true {
			color: inherit;
		}
		image: inherit;
	}
}
```

**What did you expect to happen?**
I expected to not get any linting errors, since the aforementioned source code is valid SCSS.

**What actually happened? Please include any error messages given to you by Sass Lint.**
```
test.scss
  6:3  warning  Property `image` appears to be spelled incorrectly  no-misspelled-properties
```

**Other comments:**
I noticed that the warning only occurs when certain properties belonging to the ``background`` namespace are used in place of ``image`` in my source code. In case it helps, I've provided an outline below of valid properties that don't cause warnings and other valid ones that do cause warnings.

The following properties don't produce any warnings:
- ``clip``
- ``color``
- ``size``

The following properties produce warnings:
- ``attachment``
- ``blend-mode``
- ``image``
- ``origin``
- ``position``
- ``position-x``
- ``position-y``
- ``repeat``

**Comment by DanPurdy** [comment]
Thanks @EricDunsworth I'll take a look, it seems to me that maybe the @if being present is causing some issues here. I will investigate.
